.github/workflows/check.yml

```diff
@@ -2,12 +2,47 @@ name: Gradle Check
 on: [pull_request]
 
 jobs:
-  precommit:
+  Get-CI-Image-Tag:
+    uses: opensearch-project/opensearch-build/.github/workflows/get-ci-image-tag.yml@main
+    with:
+      product: opensearch
+
+  precommit-linux:
+    needs: Get-CI-Image-Tag
+    if: github.repository == 'opensearch-project/custom-codecs'
+    runs-on: ubuntu-latest
+    container:
+      # using the same image which is used by opensearch-build team to build the OpenSearch Distribution
+      # this image tag is subject to change as more dependencies and updates will arrive over time
+      image: ${{ needs.Get-CI-Image-Tag.outputs.ci-image-version-linux }}
+      # need to switch to root so that github actions can install runner binary on container without permission issues.
+      options: --user root
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
```

Comment:
Do we need this step? Our CI image has all JDKs pre-installed

Reply:
Discussed offline: we should no alter our CI image since the goal is to use the same env

Reply:
Adding a matrix as well and will use -Druntime.java=